### PR TITLE
Change: Re-add and rework recycling for individual nodes

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

- Add new undocumented API method for recycling a single node
- Uncomment the old NodeActionMenu and bring it up to date
- Adjust types and props to handle passing around the node ID
- Make useDialog's entityID `string | number`. Bit of chicanery involved.

### CODE REVIEW

File: packages/api-v4/src/kubernetes/nodePools.ts [hunk 1]
```diff
@@ -83,3 +83,14 @@ export const recycleAllNodes = (clusterID: number, nodePoolID: number) =>
     setMethod('POST'),
     setURL(`${API_ROOT}/lke/clusters/${clusterID}/pools/${nodePoolID}/recycle`)
   );
+
+/**
+ * recycleNode
+ *
+ * Recycles a single nodes by id.
```
Comment (DevDW) [inline packages/api-v4/src/kubernetes/nodePools.ts:90]: *node

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodeActionMenu.tsx [hunk 2]
```diff
@@ -1,39 +1,48 @@
...
-  return (
+  return !matchesSmDown ? (
+    // eslint-disable-next-line
+    <>
+      {actions.map(action => {
+        return (
+          <InlineMenuAction
+            key={action.title}
+            actionText={action.title}
+            onClick={action.onClick}
```
Comment (DevDW) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodeActionMenu.tsx:38]: We should pass in `disabled={action.disabled}` here as well

**Review by johnwcallahan — APPROVED** [review]
This looks good. One UX thing: there's no feedback that a recycle is successful (except for the modal closing). Since we can't update the Node label/status right away, maybe a toast is necessary?

**Review by DevDW — COMMENTED** [review]
The functionality looks good. Left a couple of small comments. Additionally, there are two row border issues I noticed:

1. in dark mode, the action column header's border color should be black as well
Screen Shot 2021-01-12 at 5 29 40 PM

2. row border misalignment 
Screen Shot 2021-01-12 at 5 30 10 PM

for the 2nd point, I had to fix that issue in a recent PR, so this might help: https://github.com/linode/manager/pull/7279/commits/30569ad97b7ed1470166c2666a1d3d2c1b8cafe0 (actionCell styling moved from row --> action menu file)

**Review by tiffwong — APPROVED** [review]
Approving pending the other feedback.
Unrelated but we should probably make another ticket to update the table to use the CMR styles

**Review by DevDW — APPROVED** [review]

**Pull request merged**